src/acquisition/flusurv/flusurv.py

```diff
@@ -120,17 +122,46 @@ def fetch_json(path, payload, call_count=1, requests_impl=requests):
     return resp.json()
 
 
-def fetch_flusurv_object(location_code):
-    """Return decoded FluSurv JSON object for the given location."""
-    return fetch_json(
-        "PostPhase03GetData",
+def fetch_flusurv_location(location, seasonids):
+    """Return FluSurv JSON object for the given location."""
+    location_code = location_to_code[location]
+
+    result = fetch_json(
+        "PostPhase03DataTool",
         {
             "appversion": "Public",
-            "networkid": location_code[0],
-            "cacthmentid": location_code[1],
+            "key": "getdata",
+            "injson": [
+                {
+                    "networkid": location_code[0],
+                    "catchmentid": location_code[1],
+                    "seasonid": elem,
+                } for elem in seasonids],
         },
     )
 
+    # If no data is returned (a given seasonid is not reported,
+    # location codes are invalid, etc), the API returns a JSON like:
+    #    {
+    #        'default_data': {
+    #            'response': 'No Data'
+    #            }
+    #    }
+    #
+    # If data is returned, then data["default_data"] is a list
+    #  and data["default_data"]["response"] doesn't exist.
+    assert isinstance(result["default_data"], list) and len(result["default_data"]) > 0, \
+        f"Data was not correctly returned from the API for {location}"
```

Comment:
should this always raise an exception?  that is, does a lack of data for a location indicate a problem on our end?

also, i think it is better to raise an explicit `Exception` here (as is done in other places in this file) instead of using the `assert` statement.

Reply:
> does a lack of data for a location indicate a problem on our end?

I don't know. Given that this has been effectively down for 2 years, I thought it would be good to over-alert rather than under-alert. Again, the API is internal to the CDC so we don't really know what normal behavior is for it.

> Exception -> assert

:+1:

Reply:
Not sure these are the cases you're talking about here, but:
- FluSurv-NET isn't always the same set of states, but the set should only change once in a while, maybe exclusively between seasons.
- On their dashboard, I only remember seeing "No Data Available." for weekly rates in 2020-21, when you get to them by first looking at the 2020-21 Cumulative Rate then switching to Weekly Rate.

Reply:
Okay, good to know that the location set does change. In that case, maybe we need to decide what locations to request based on what the API metadata dynamically says is available. Right now we use a hard-coded list, which means new locations would not be requested on our side. This means that erroring for missing locations also might be overkill -- maybe will switch this to a warning.

On the missing 2020-2021 data, I think this is a dashboard issue. I'll double-check, but the API response seems to include data fro that range.

Reply:
I modified this so that we only request data for locations that are reported in the metadata to be available for recent dates. So if we don't get any data back, that actually is a problem.